### PR TITLE
[Truffle] Adding frozen check to Array#delete.

### PR DESCRIPTION
**Pull request by bjfish**

### CODE REVIEW

File: truffle/src/main/java/org/jruby/truffle/nodes/core/ArrayNodes.java [hunk 1]
```diff
@@ -800,10 +803,17 @@ public DeleteNode(RubyContext context, SourceSection sourceSection) {
...
         }
 
         @Specialization(guards = "isIntegerFixnum")
         public Object deleteIntegerFixnum(VirtualFrame frame, RubyArray array, Object value) {
+            if (isFrozenNode == null) {
+                CompilerDirectives.transferToInterpreter();
+                isFrozenNode = insert(IsFrozenNodeFactory.create(getContext(), getSourceSection(), null));
+            }
+            final boolean isFrozen = isFrozenNode.executeIsFrozen(array);
+
```
Comment (chrisseaton) [inline truffle/src/main/java/org/jruby/truffle/nodes/core/ArrayNodes.java]: Can't this be hoisted to line 827?

**Comment by chrisseaton** [comment]
Could this be another annotation?

**Comment by bjfish** [comment]
@chrisseaton I've hoisted that and refactored this a little.

What did you have in mind for an annotation? I don't see this particular case occurring frequently.

**Comment by chrisseaton** [comment]
Ok, if I find myself copying and pasting this I'll think about it.

**Pull request merged**